dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report-sql.repo.ts

```diff
@@ -1,10 +1,35 @@
 // TODO FN-1853 - rename this to `utilisation-report.repo.ts` when all repo
 //  methods have been migrated from MongoDB to SQL
 import { SqlDbDataSource } from '@ukef/dtfs2-common/sql-db-connection';
-import { UtilisationReportEntity, ReportPeriod } from '@ukef/dtfs2-common';
+import { UtilisationReportEntity, ReportPeriod, UTILISATION_REPORT_RECONCILIATION_STATUS } from '@ukef/dtfs2-common';
+
+export type GetUtilisationReportDetailsOptions = {
+  reportPeriod?: ReportPeriod;
+  excludeNotUploaded?: boolean;
+};
 
 export const UtilisationReportRepo = SqlDbDataSource.getRepository(UtilisationReportEntity).extend({
   async findOneByBankIdAndReportPeriod(bankId: string, reportPeriod: ReportPeriod): Promise<UtilisationReportEntity | null> {
     return await this.findOneBy({ bankId, reportPeriod });
   },
+
+  async findAllByBankId(bankId: string, options?: GetUtilisationReportDetailsOptions): Promise<UtilisationReportEntity[]> {
```

Comment:
This method is quite involved and I am aware I have not written any tests for it, whilst I have deleted the unit tests relating to the previous mongo implementation. I'm not sure what unit tests if any we want on SQL repositories like this or if we want to cover with api/e2e tests since we have the db connection?

Reply:
Good question, I'm sure we could figure out how to write some tests where we mock the createQueryBuilder and check it's made in the right way but can also imagine it might be a little grim. If we have the coverage of the different if loops in API tests which I've seen you've added we should be fine without :)